linkerd/proxy/http/src/version.rs

```diff
@@ -21,12 +21,11 @@ impl std::convert::TryFrom<http::Version> for Version {
     }
 }
 
-// A convenience for tracing contexts.
-impl std::fmt::Display for Version {
+impl std::fmt::Debug for Version {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            Self::Http1 => write!(f, "1.x"),
-            Self::H2 => write!(f, "h2"),
+            Self::Http1 => write!(f, "HTTP/1"),
+            Self::H2 => write!(f, "HTTP/2"),
```

Comment:
_technically_ this feels more correct as a `Display` impl to me, since I feel like the general idiom for `Debug` is to mimic the way the type appears in Rust source code...but if this was the `Display` impl, we would just basically never use the `Debug` impl, and always using this formatting seems fine to me, so it's not a big deal. probably better to just use this formatting consistently.

Reply:
FWIW this is what the `http` crate's `Version` does -- i'm assuming because you'll never see `HTTP/2` anywhere in a request stream...